e2e/Utils.ts

```diff
@@ -83,6 +84,7 @@ export const writeFiles = (
       createDirectory(path.join.apply(path, [directory].concat(filePath)));
     }
     fs.writeFileSync(
+      // @ts-ignore
```

Comment:
why the ignore? The code looks really convoluted (I guess it's from a time before rest params), could we do

```ts
path.resolve(directory, ...filePath, filename),
```
?

Same above, just do `path.join(directory, ...filePath)`

Reply:
Oh the issue here was that filename is can possibly undefined from the `pop` call. It doesnt fir the type expected by `path.resolve`

Reply:
We could do this to satisfy the linter but just seemed hacky to me. 
```javascript 
path.resolve(directory, ...filePath, (filename ? filename : ""))
```
what do you think?

Reply:
I've pushed up some changes to make the code simpler regarding your comments. I'll just continue tomorrow. It's midnight where I'm at.  Happy coding :)